3015-Delete-user-profile-and-change-password - Add user deletion feature

RockTF opened the pull request:
[3015-Delete-user-profile-and-change-password](https://github.com/ita-social-projects/GreenCity/issues/3015)

Introduce functionality to deactivate users by setting their status to DELETED. Added corresponding methods in the service and controller layers, updated security configurations, and extended unit tests to cover this new feature.

IliaWithHat committed:
Deletion of unused code.

IliaWithHat committed:
Merge branch 'dev' into remove-unused-code

IliaWithHat committed:
SonarCloud fix

IliaWithHat committed:
Merge remote-tracking branch 'origin/remove-unused-code' into remove-unused-code

RockTF committed:
3015-Delete-user-profile-and-change-password - Add user deletion feature

Introduce functionality to deactivate users by setting their status to DELETED. Added corresponding methods in the service and controller layers, updated security configurations, and extended unit tests to cover this new feature.

RockTF committed:
3015-Delete-user-profile-and-change-password - Add user deletion feature

Introduce functionality to deactivate users by setting their status to DELETED. Added corresponding methods in the service and controller layers, updated security configurations, and extended unit tests to cover this new feature.

RockTF committed:
3015-Delete-user-profile-and-change-password - Add user deletion feature

Introduce functionality to deactivate users by setting their status to DELETED. Added corresponding methods in the service and controller layers, updated security configurations, and extended unit tests to cover this new feature.

RockTF committed:
3015-Delete-user-profile-and-change-password - Add user deletion feature

Introduce functionality to deactivate users by setting their status to DELETED. Added corresponding methods in the service and controller layers, updated security configurations, and extended unit tests to cover this new feature.

RockTF committed:
3015-Delete-user-profile-and-change-password - Format code comments and align indentation

Refactor code comments for clarity and adjust indentation for consistency. This change improves the readability of the codebase, making it easier for developers to understand the purpose of various methods. Indentation alignment also ensures a clean and maintainable code structure.

RockTF committed:
3015-Delete-user-profile-and-change-password - Format code comments and align indentation

Refactor code comments for clarity and adjust indentation for consistency. This change improves the readability of the codebase, making it easier for developers to understand the purpose of various methods. Indentation alignment also ensures a clean and maintainable code structure.

RockTF committed:
3015-Delete-user-profile-and-change-password - Format code comments and align indentation

Refactor code comments for clarity and adjust indentation for consistency. This change improves the readability of the codebase, making it easier for developers to understand the purpose of various methods. Indentation alignment also ensures a clean and maintainable code structure.

RockTF committed:
3015-Delete-user-profile-and-change-password - Format code comments and align indentation

Refactor code comments for clarity and adjust indentation for consistency. This change improves the readability of the codebase, making it easier for developers to understand the purpose of various methods. Indentation alignment also ensures a clean and maintainable code structure.

RockTF committed:
3015-Delete-user-profile-and-change-password - Remove redundant import statement.

The unused static import statement was deleted from OwnSecurityController.java. This change helps in maintaining a clean and readable codebase by ensuring that only necessary imports are included.

RockTF committed:
Merge remote-tracking branch 'origin/3015-Delete-user-profile-and-change-password' into 3015-Delete-user-profile-and-change-password

# Conflicts:
#	core/src/main/java/greencity/security/controller/OwnSecurityController.java

KizerovDmitriy commented on core/src/main/java/greencity/security/controller/OwnSecurityController.java:
```diff
@@ -42,14 +43,7 @@
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.ResponseStatus;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
```
wildcard

KizerovDmitriy commented on core/src/test/java/greencity/security/controller/OwnSecurityControllerTest.java:
```diff
@@ -10,23 +10,22 @@
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
we don't use wildcard imports

KizerovDmitriy commented:
Check sonarcloud issues

IliaWithHat committed:
Merge pull request #376 from ita-social-projects/remove-unused-code

Remove unused code

RockTF committed:
3015-Delete-user-profile-and-change-password - Refactor import statements in OwnSecurityController.

Separated import statements for clarity and organization. This change improves readability and maintainability of both the controller and its test file.

RockTF committed:
3015-Delete-user-profile-and-change-password - Refactor import statements in OwnSecurityController.

Separated import statements for clarity and organization. This change improves readability and maintainability of both the controller and its test file.

RockTF committed:
3015-Delete-user-profile-and-change-password -
Reorganize and clean up security test imports

Reorganized import statements for consistent and readable structure. Removed the unnecessary initMocks call in OwnSecurityServiceImplTest to streamline the setup for Mockito.

RockTF committed:
Merge remote-tracking branch 'origin/3015-Delete-user-profile-and-change-password' into 3015-Delete-user-profile-and-change-password

# Conflicts:
#	core/src/main/java/greencity/security/providers/JwtAuthenticationProvider.java
#	service/src/test/java/greencity/security/service/OwnSecurityServiceImplTest.java
